src/host/readDataCooked.cpp

```diff
@@ -605,15 +680,13 @@ void COOKED_READ_DATA::_handleVkey(uint16_t vkey, DWORD modifiers)
     }
 }
 
-#undef _buffer
-#undef _bufferCursor
-
 // Handles any tasks that need to be completed after the read input loop finishes,
 // like handling doskey aliases and converting the input to non-UTF16.
 void COOKED_READ_DATA::_handlePostCharInputLoop(const bool isUnicode, size_t& numBytes, ULONG& controlKeyState)
 {
     auto writer = _userBuffer;
-    std::wstring_view input{ _buffer };
+    auto buffer = _buffer.Extract();

```

Comment:
this code didn't used to destroy `_buffer` on all codepaths. now it does! is it OK to do so?

Reply:
Yes, because this function can only be called once during the lifetime of the object instance.